capsules/src/can.rs

```diff
@@ -0,0 +1,486 @@
+            8 => {
+                self.appid.set(appid);
+                self.can_rx.take().map(|dest_buffer| {
+                    self.apps.enter(appid, |_, kernel| {
+                        match kernel.get_readwrite_processbuffer(0).map_or_else(|err| err.into(), |buffer_ref| {
+                            buffer_ref.enter(|buffer| {
+                                if buffer.len() >= 16 + size_of::<u32>() {
+                                    Ok(())
+                                } else {
+                                    Err(ErrorCode::SIZE)
+                                }
+                            }).unwrap_or_else(|err| err.into())
+                        }) {
+                            Ok(_) => {
+                                match self.can.start_receive_process(dest_buffer) {
+                                    Ok(_) => CommandReturn::success(),
+                                    Err((err, _)) => CommandReturn::failure(err),
+                                }
+                            },
+                            Err(err) => CommandReturn::failure(err.into()),
+                        }
+                    }).unwrap_or_else(|err| err.into())
+                }).unwrap_or(CommandReturn::failure(ErrorCode::NOMEM))
+            }
+
+            // Stop receiving messages
+            9 => {
+                self.appid.set(appid);
+                match self.can.stop_receive() {
+                    Ok(_) => CommandReturn::success(),
+                    Err(err) => CommandReturn::failure(err),
+                }
+            }
+            _ => CommandReturn::failure(ErrorCode::NOSUPPORT),
+        }
+    }
+
+    fn allocate_grant(&self, process_id: ProcessId) -> Result<(), kernel::process::Error> {
+        self.apps.enter(process_id, |_, _| {})
+    }
+}
+
+impl<'a, Can: can::Can> can::ControllerClient for CanCapsule<'a, Can> {
+    // This callback must be called after an `enable` or `disable` command was sent.
+    // If the new state of the peripheral received as an argument is different than
+    // the state the capsule expects, send to the userspace an error callback. If
+    // the state is the right one, wait for the next callback of the process (the
+    // `enabled` for the `enable` command and `disabled` for the `disable` command).
+    fn state_changed(&self, state: can::State) {
+        match self.wait_for_state.map(|capsule_state| -> (can::State, u8) {
+            capsule_state.1 = capsule_state.1 + 1;
+            if state != capsule_state.0 {
+                self.error_occured.set(true);
+            }
+            *capsule_state
+        }) {
+            Some(state) => self.wait_for_state.set(state),
+            None => {
+                if self.error_occured.is_some() {
+                    match state {
+                        can::State::Running | can::State::Error(_) => {
+                            self.schedule_callback(2, ErrorCode::FAIL as usize, 0, 0)
```

Comment:
Use the define constant.
```suggestion
                            self.schedule_callback(upcalls::UPCALL_MEANING_2, ErrorCode::FAIL as usize, 0, 0)
```